src/rpc/budget.cpp

```diff
@@ -48,14 +48,53 @@ void budgetToJSON(CBudgetProposal* pbudgetProposal, UniValue& bObj)
     bObj.push_back(Pair("fValid", pbudgetProposal->fValid));
 }
 
-UniValue preparebudget(const UniValue& params, bool fHelp)
+void checkBudgetInputs(const UniValue& params, std::string &strProposalName, std::string &strURL,
+                       int &nPaymentCount, int &nBlockStart, CBitcoinAddress &address, CAmount &nAmount)
 {
     int nBlockMin = 0;
     CBlockIndex* pindexPrev = chainActive.Tip();
 
+    if (strProposalName.size() > 20)
+        throw JSONRPCError(RPC_INVALID_PARAMETER, "Invalid proposal name, limit of 20 characters.");
+
+    strURL = SanitizeString(params[1].get_str());
+    std::string strErr;
+    if (!validateURL(strURL, strErr))
+        throw JSONRPCError(RPC_INVALID_PARAMETER, result);
```

Comment:
`result` is undefined

Reply:
oh crap; that's what happens when I make "one last change!" while my kids are trying to get my attention :)

Reply:
I can totally relate :stuck_out_tongue_closed_eyes:

Reply:
cleaned up and build tested